Add prettier, eslint, husky and lintstaged

Johnnycus opened the pull request:
ESLint suggests replacing `for...in` with array loops so I added `// TODO` comments there. Or we can turn that rule off.

billythedummy commented:
> ESLint suggests replacing `for...in` with array loops so I added `// TODO` comments there. Or we can turn that rule off.

Lets keep that rule, there seems to be only 1 instance where `for (... in ...)` is used and thats in `reverse()` in `stake-pool/utils.ts`. Could you refactor that to iterate `Object.keys()` instead? Thanks!

Johnnycus commented:
> > ESLint suggests replacing `for...in` with array loops so I added `// TODO` comments there. Or we can turn that rule off.
> 
> Lets keep that rule, there seems to be only 1 instance where `for (... in ...)` is used and thats in `reverse()` in `stake-pool/utils.ts`. Could you refactor that to iterate `Object.keys()` instead? Thanks!

Done, there were also some instances of `for...of` which I also replaced with `forEach`. I only kept one `for...of` that had `async/await`. We can replace it with `map`, but this will run the, in parallel

billythedummy commented on .eslintrc.json:
```diff
@@ -0,0 +1,34 @@
+    "@typescript-eslint/no-use-before-define": "warn",
+    "no-plusplus": "off",
+    "import/no-cycle": "warn",
+    "no-await-in-loop": "off",
```
think we can keep this on and just use eslint disable comments when we need to break the rule like what you did in signAndSendTransactionSequence?

Johnnycus replied:
Agreed, will update

billythedummy reviewed (approved):
there's gonna be a lot to add once https://github.com/igneous-labs/stake-pool-sdk/pull/36 is merged, sorry about that
